PMM-12432_Release_Notes_2.40.0 (#1140)
percona/pmm-doc: docs/release-notes/2.40.0.md

--->

## Release Highlights

### Inventory improvements

#### Ability to update Labels for already added Services to PMM  
PMM-11423,PMM-11148 : In the past, the only solution to fixing incorrect labels was to remove and re-add the Service to PMM. However, now it is possible to edit the instance labels directly in the UI without needing to go through that process. This update also includes an improved UI for adding remote Databases to PMM.
[IMAGE HERE]

#### Connecting Services and Nodes  
PMM-11963: It used to be difficult to figure out which Service was operating on which Node in PMM, but now things are easier. This new release lets you view a list of Services running on Nodes, along with the Node's name where the Service is being used. You can even filter by these names to make it simpler to navigate.
[IMAGE HERE]

#### Cluster view for Inventory [Technical Preview]
PMM-12378: Understanding the structure of  your Inventory is important. In this release, we starting the experimental feature with the ability to see all your Services grouped by Cluster so users can understand inventory more easily. 
[IMAGE HERE]

 

### MongoDB backup monitoring 

> PMM-12384: Now, PMM not only allows you to perform MongoDB backups from the UI, but it also displays backup status metrics. Additionally, we are shipping the first version of the Alert Template to notify you of any failed backups.

Reviewer: It would be useful to add a link to the documentation of the new backup status metrics. I guess this doc will be included as part of https://jira.percona.com/browse/PMM-12484. I think important to link here, so users can learn more. Thanks!

| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

<!---

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

--->

## Release Highlights

### Inventory improvements

#### Ability to update labels for existing services to PMM  

Starting with PMM 2.40.0, editing labels for existing services is easier and much more efficient. You can edit a service label directly from the PMM UI without removing and re-adding it. For more information on editing labels, see [documentation](https://docs.percona.com/percona-monitoring-and-management/details/dashboards/dashboard-inventory.html#editing-labels-for-a-service).

>   ![!](../../_images/PMM_access_edit_labels.png)

Reviewer 2: I think I'm getting 40 for this image

Author: This image is used in another PR. so when all the PR's are merged the image will be visible.

  ![!image](../../_images/PMM_Inventory_cluster_view.png)

For detailed information on this feature, see [documentation](https://docs.percona.com/percona-monitoring-and-management/details/dashboards/dashboard-inventory.html#cluster-view)
 

### MongoDB Backup Monitoring 

With this release, we are shipping the first version of a the **Backup failed** alert template which notifies of any failed MongoDB backups. This uses a new, dedicated metric, `pmm_managed_backups_artifacts`, for checking the state of backup artifacts. 

This new template is currently in Technical Preview and we are keen to get your feedback around this change before we move it to General Availability.

  ![!image](../_images/Backup_Failed_Template.png) 

For information on working with this new template, see the [Percona Alerting](../get-started/alerting.md) topic.

### Components Upgrade

VictoriaMetrics has been upgraded [v1.93.4](https://docs.victoriametrics.com/CHANGELOG.html#v1934).

## New Features

> - [PMM-9374](https://jira.percona.com/browse/PMM-9374) - Starting with PMM 2.40.0, you can now use an external VictoriaMetrics database for monitoring in PMM. This provides multiple benefits, including scalability, resource isolation, reduced load on the PMM server,  customization, etc.

Reviewer 2: this needs to be marked as technical preview and be at the bottom of the section

Author: Done

Author: Done.

    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

--->

## Release Highlights

### Inventory improvements

#### Ability to update labels for existing services to PMM  

Starting with PMM 2.40.0, editing labels for existing services is easier and much more efficient. You can edit a service label directly from the PMM UI without removing and re-adding it. For more information on editing labels, see [documentation](https://docs.percona.com/percona-monitoring-and-management/details/dashboards/dashboard-inventory.html#editing-labels-for-a-service).

  ![!](../../_images/PMM_access_edit_labels.png)

Furthermore, in our continued effort  to enhance user experience, we have redesigned the following pages to make them more user-friendly and intuitive:

- Select service type page
- Add service page

#### Connecting Services and Nodes 

Starting with PMM 2.40.0, you can click on the link in the **Node Name** column to view the node on which a specific service is running and analyze how node-level resource utilization impacts the performance of those services.

> 

Reviewer 2: This also links from the Node page to Services running on this Node. will be good to describe and image

Author: Done.

  ![!](../../_images/PMM_Inventory_service_node_relation.png)

#### Cluster view for Inventory [Technical Preview]

Understanding the structure of your inventory is crucial. With the release of PMM 2.40.0, we've introduced an experimental feature that categorizes all your Services by Cluster, making it easier for you to understand your inventory. By using the **Organize by Clusters** toggle, you can view a group of services as a single cluster. PMM utilizes the cluster label to display services that belong to the same cluster.

  ![!image](../../_images/PMM_Inventory_cluster_view.png)

For detailed information on this feature, see [documentation](https://docs.percona.com/percona-monitoring-and-management/details/dashboards/dashboard-inventory.html#cluster-view)
 

### MongoDB backup monitoring 
PMM-12384: Now, PMM not only allows you to perform MongoDB backups from the UI, but it also displays backup status metrics. Additionally, we are shipping the first version of the Alert Template to notify you of any failed backups.

> [IMAGE HERE]

Reviewer 3: Missing image

> For the usage of this Template, please use Alert documentation (LINK)

Reviewer 3: And link

- [PMM-12457](https://jira.percona.com/browse/PMM-12457) - We have added a new `node_name` property to the **PMM agent down** alert template in PMM. This allows users to easily identify the node where the failure occurred.
- [PMM-12488](https://jira.percona.com/browse/PMM-12488) - VictoriaMetrics has been updated to v1.93.4.
- [PMM-12500](https://jira.percona.com/browse/PMM-12500) - [Tech Preview] K8s monitoring for Percona Operator users - basic dashboard

## Bugs Fixed

- [PMM-10145](https://jira.percona.com/browse/PMM-10145) - When we installed an AMI image in AWS and added an Amazon RDS instance in the creating state, it caused errors. The issue has been resolved now.
- [PMM-12173](https://jira.percona.com/browse/PMM-12173) - On adding several clients to PMM, unexpected and unauthorized errors caused PMM to stop responding. The issue has since been resolved now.

> - [PMM-12221](https://jira.percona.com/browse/PMM-12221) - Fixed an issue where the user was logged out during an upgrade.

Reviewer 4: this one also can be removed

- [PMM-12344](https://jira.percona.com/browse/PMM-12344) - An error was displayed on the **Explain** tab after selecting a PostgreSQL query. The issue has been resolved now.
- [PMM-12361](https://jira.percona.com/browse/PMM-12361) - The command `pmm-admin inventory` returned GO errors. The issue has been reolved now.
- [PMM-12382](https://jira.percona.com/browse/PMM-12382) - Fixed an issue where the upper-case custom labels were not being accepted.
- [PMM-12420](https://jira.percona.com/browse/PMM-12420) - PMM upgrade was failing due to migration from SQLite to PGSQL. The issue has been resolved now.

> - [PMM-11371](https://jira.percona.com/browse/PMM-11371) - pmm-admin config failed to register agent to pmm-server. The issue has been resolved now.

Reviewer 5: ```suggestion
- [PMM-11371](https://jira.percona.com/browse/PMM-11371) - Usage of pmm-admin config shutdown pmm-agent. The issue has been resolved now.
```

- [PMM-9374](https://jira.percona.com/browse/PMM-9374) - Starting with PMM 2.40.0, you can now use an external VictoriaMetrics database for monitoring in PMM. This provides multiple benefits, including scalability, resource isolation, reduced load on the PMM server,  customization, etc.
- [PMM-11963](https://jira.percona.com/browse/PMM-11963) - Starting with PMM 2.40.0, you can click on the link in the **Node Name** column to view the node on which a specific service is running.
- [PMM-11148](https://jira.percona.com/browse/PMM-11148) - [Inventory]: We have redesigned the **Select service type** and **Add service** pages to make them more user-friendly and intuitive.
- [PMM-11423](https://jira.percona.com/browse/PMM-11423) - [Inventory]: Starting with PMM 2.40.0, you can now edit service labels directly from the PMM UI without having to remove and re-add them.
- [PMM-12378](https://jira.percona.com/browse/PMM-12378) - [Inventory] - We have introduced an experimental feature that categorizes all your Services by Cluster, making it easier for you to understand your inventory. By using the **Organize by Clusters** toggle, you can view a group of services as a single cluster. 

> - [PMM-12384](https://jira.percona.com/browse/PMM-12378) - [BE] Backup metrics and alert rule template

Reviewer 6: I don't know what is the right ticket for this, but this one is not :)
Also, the description seems to not mention what has been done.

Author: Fixed.

## Improvements

- [PMM-4466](https://jira.percona.com/browse/PMM-4466) - Grafana now uses PostgreSQL instead of SQLite, resulting in improved performance, stability for user auth requests, data integrity, reliability, security, etc.
- [PMM-12310](https://jira.percona.com/browse/PMM-12310) - Links from PMM to documentation may change, causing "broken links" in older PMM versions due to document structure changes. We have replaced all links with Short.io links specific to each document to address this. This way, we can maintain the PMM-to-doc links using a URL shortener. This ensures that the links remain accessible even if the document structure changes.

> - [PMM-12457](https://jira.percona.com/browse/PMM-12457) - We have updated the **PMM agent down** alert template in PMM with a new `node_name` property. This allows users to easily identify the node where the failure occurred.

Reviewer 6: ```suggestion
- [PMM-12457](https://jira.percona.com/browse/PMM-12457) - We have added a new `node_name` property to the **PMM agent down** alert template in PMM. This allows users to easily identify the node where the failure occurred.
```

- [PMM-12500](https://jira.percona.com/browse/PMM-12500) - [Tech Preview] K8s monitoring for Percona Operator users - basic dashboard

## Bugs Fixed

- [PMM-10145](https://jira.percona.com/browse/PMM-10145) - When we installed an AMI image in AWS and added an Amazon RDS instance in the creating state, it caused errors. The issue has been resolved now.
- [PMM-12173](https://jira.percona.com/browse/PMM-12173) - On adding several clients to PMM, unexpected and unauthorized errors caused PMM to stop responding. The issue has since been resolved now.
- [PMM-12221](https://jira.percona.com/browse/PMM-12221) - Fixed an issue where the user was logged out during an upgrade.
- [PMM-12344](https://jira.percona.com/browse/PMM-12344) - An error was displayed on the **Explain** tab after selecting a PostgreSQL query. The issue has been resolved now.
- [PMM-12361](https://jira.percona.com/browse/PMM-12361) - The command `pmm-admin inventory` returned GO errors. The issue has been reolved now.
- [PMM-12382](https://jira.percona.com/browse/PMM-12382) - Fixed an issue where the upper-case custom labels were not being accepted.

> - [PMM-12420](https://jira.percona.com/browse/PMM-12420) - PMM upgrade was failing due to migration from SQLite to PGSQL. The issue has been resolved now.

Reviewer 4: let's remove this item from Release notes

> - [PMM-12420](https://jira.percona.com/browse/PMM-12420) - PMM upgrade was failing due to migration from SQLite to PGSQL. The issue has been resolved now.

Author: Done

- [PMM-11371](https://jira.percona.com/browse/PMM-11371) - pmm-admin config failed to register agent to pmm-server. The issue has been reolved now.
- [PMM-11603](https://jira.percona.com/browse/PMM-11603) - When running `pmm-agent` outside of systemd, adding an agent causes errors. The issue has been resolved now.
- [PMM-11651](https://jira.percona.com/browse/PMM-11651) - When a user has both full access and limited access roles, only the limited role's metrics are displayed. The issue has been resolved now.

> - [PMM-12146](https://jira.percona.com/browse/PMM-12146) - Dead Tuples graph on PostgreSQL Vacuum Monitoring Dashboard displayed invalid percentage. The 

Author: Fixed.

> - [PMM-12448](https://jira.percona.com/browse/PMM-12448) - Can't start backup if artifact with empty service_id exists.

Reviewer 6: Let's maybe tell what has been done here )

Author: This will be worked upon by @catalina-adam

Author: Done

> - [PMM-12494](https://jira.percona.com/browse/PMM-12494) - While trying to upgrade PMM using the docker, an error was being thrown. The issue has been reolved now.

Reviewer 6: ```suggestion
- [PMM-12494](https://jira.percona.com/browse/PMM-12494) - While trying to upgrade PMM using the docker, an error was being thrown. The issue has been resolved now.
```
